contracts/chain-adapters/Solana_Adapter.sol

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { IMessageTransmitter, ITokenMessenger } from "../external/interfaces/CCTPInterfaces.sol";
+import { SpokePoolInterface } from "../interfaces/SpokePoolInterface.sol";
+import { AdapterInterface } from "./interfaces/AdapterInterface.sol";
+import { CircleCCTPAdapter, CircleDomainIds } from "../libraries/CircleCCTPAdapter.sol";
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/**
+ * @notice Contract containing logic to send messages from L1 to Solana.
+ * @dev Public functions calling external contracts do not guard against reentrancy because they are expected to be
+ * called via delegatecall, which will execute this contract's logic within the context of the originating contract.
+ * For example, the HubPool will delegatecall these functions, therefore it's only necessary that the HubPool's methods
+ * that call this contract's logic guard against reentrancy.
+ * @custom:security-contact bugs@across.to
+ */
+
+// solhint-disable-next-line contract-name-camelcase
+contract Solana_Adapter is AdapterInterface, CircleCCTPAdapter {
```

Comment:
nit: is this a "solana_adapter" or it is better called "Cctp_Adapter"?

Reply:
This includes logic specific to Solana, e.g. how `setEnableRoute` is translated. I was though considering to move `cctpMessageTransmitter` to base `CircleCCTPAdapter`, but that would have required adding it to constructor for all other adapters that don't use CCTP for general message transmission.